### ISSUE TITLE
Nice-to-have feature: Allow to specify an additional indent amount

### ISSUE THREAD

**Issue by mlegner** [dtolnay/indoc#63]
I'm using `indoc` to generate a YAML file. This is done piecewise; among other reasons to generate an array through a loop. Now these pieces need to be indented to produce a valid YAML.

It would be really nice to be able to specify an additional indentation amount that would apply to all lines. The effective code would be a minor change in https://github.com/dtolnay/indoc/blob/master/src/unindent.rs#L45-L47, but this would of course require more changes to pass around the parameter.

Currently, I see two workarounds for my use case:
1. Use a dummy character on the first line to define the indent level, which is either ignored (e.g., a line comment) or later removed. (This is what I do currently.)
2. Manually add spaces to the lines afterwards. Unfortunately this somewhat defeats the use of `indoc` in the first place.

I would be happy to contribute this feature, but I wanted to check if this is even desired before starting on it.

**Comment by dtolnay** [comment]
I would not want this built into this crate. But it would be nice for some to maintain a more fully featured macro library which could serve this use case.
